packages/fxa-content-server/app/tests/spec/views/settings/display_name.jsx

```diff
@@ -0,0 +1,203 @@
+import TestHelpers from '../../../lib/helpers';
+import User from 'models/user';
+import View from 'views/settings/display_name';
+import {ChangeorAddButtonComponent, DisplayNameFormComponent, DisplayNameComponent} from 'views/settings/display_name';
+import { render, cleanup, fireEvent, queryByText } from '@testing-library/react';
+
+describe('views/settings/display_name', function () {
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('ChangeorAddButtonComponent', () => {
+    function renderButtonComponent (displayName = ''){
+      render(
+        <ChangeorAddButtonComponent displayName={displayName}/>
+      );
+    }
+
+    it('renders the button correctly without displayName set', () => {
+      renderButtonComponent('');
+      const settingsButtons = document.querySelectorAll('.settings-button');
+      assert.lengthOf(settingsButtons, 1);
+      assert.include(settingsButtons[0].innerText, 'Add');
+    });
+
+    it('renders the button correctly with a displayName set', () => {
+      renderButtonComponent('joe');
+
+      const settingsButtons = document.querySelectorAll('.settings-button');
+      assert.lengthOf(settingsButtons, 1);
+      assert.include(settingsButtons[0].innerText, 'Change');
+    });
+  });
+
+  describe('DisplayNameComponent', () => {
+    function renderDisplayNameComponent (){
+      render(
+        <DisplayNameComponent />
+      );
+    }
+
+    it('renders correctly', () => {
+      renderDisplayNameComponent();
+
+      const headerEls = document.querySelectorAll('.settings-unit-summary');
+      assert.lengthOf(headerEls, 1);
+      assert.include(headerEls[0].innerText, 'Display name');
+    });
+  });
+
+  describe('DisplayNameFormComponent', ()=>{
+    var account;
+    var email;
+    var user;
+    beforeEach(function () {
+      email = TestHelpers.createEmail();
+      user = new User();
+      account = user.initAccount({
+        email: email,
+        sessionToken: 'fake session token',
+        uid: 'uid',
+        verified: true
+      });
+    });
+    function renderDisplayNameFormComponent (displayName = ''){
+      render(
+        <DisplayNameFormComponent displayName={displayName} account={account}/>
+      );
+    }
+
+    it('validates the display name field for changes:name did not change',() => {
+      account.set('displayName', 'joe');
+      renderDisplayNameFormComponent('joe');
+
+      const submitDisplay = document.querySelectorAll('#submit_display');
+      assert.lengthOf(submitDisplay, 1);
+      assert.equal(submitDisplay[0].getAttribute('disabled'),'','name did not change');
+    });
+
+    it('validates the display name field for changes:name changed',() => {
+      account.set('displayName', 'joe');
+      renderDisplayNameFormComponent('joe changed');
+
+      const submitDisplay = document.querySelectorAll('#submit_display');
+      assert.lengthOf(submitDisplay, 1);
+      assert.equal(submitDisplay[0].attributes.disabled,null,'name changed');
+    });
```

Comment:
Is there a way to test form submission and ensure the expected function is called with the expected parameters?